cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java

```diff
@@ -168,6 +168,56 @@ public void configure(Map<String, ?> config) {
     ensureTopicsCreated(config);
   }
 
+  /**
+   * Retrieve the desired replication factor of sample store topics.
+   *
+   * @param config The configurations for Cruise Control.
+   * @param adminClient The adminClient to send describeCluster request.
+   * @return Desired replication factor of sample store topics, or {@link null} if failed to resolve replication factor.
+   */
+  protected Short sampleStoreTopicReplicationFactor(Map<String, ?> config, AdminClient adminClient) {
+    if (_sampleStoreTopicReplicationFactor == null) {
+      short numberOfBrokersInCluster;
+      try {
+        numberOfBrokersInCluster = (short) adminClient.describeCluster().nodes().get(CLIENT_REQUEST_TIMEOUT_MS,
+                                                                                     TimeUnit.MILLISECONDS).size();
+      } catch (InterruptedException | ExecutionException | TimeoutException e) {
+        LOG.warn("Auto creation of sample store topics failed due to failure to describe cluster.", e);
+        return null;
+      }
+      if (numberOfBrokersInCluster <= 1) {
+        throw new IllegalStateException(String.format("Kafka cluster has less than 2 brokers (brokers in cluster=%d, zookeeper.connect=%s)",
+                                                      numberOfBrokersInCluster, config.get(ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG)));
+      }
+
+      return (short) Math.min(DEFAULT_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR, numberOfBrokersInCluster);
```

Comment:
if we reaches here `numberOfBrokersInCluster` is guaranteed to be equal or larger than `DEFAULT_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR`(which is 2), looks the  `Math.min()` is redundant here.

Reply:
Doesn't this assume that `DEFAULT_SAMPLE_STORE_TOPIC_REPLICATION_FACTOR` will always be `2`? -- i.e. would it still be correct if we change the default RF to `3` or make it configurable in the future?